### PR TITLE
Add Prerequisite and Build source code section

### PR DESCRIPTION
**Pull request by ykgarfield**

### CODE REVIEW

**Comment by tisonkun** [comment]
ref #1

**Review by tisonkun — CHANGES_REQUESTED** [review]
Thanks for your contribution @ykgarfield! The content looks good to me.

However, I think it would be better if we convert the descriptive guide into a manual-like guide. That is, instead of document it in long sentence, we list out the prerequisite by items and build guide step by step.

You can take a look at [the corresponding section](https://github.com/apache/flink#building-apache-flink-from-source) in Apache Flink or [one of my personal project](https://github.com/tisonkun/trufflebf#prerequisites)

**Comment by gosonzhang** [comment]
Thanks

**Review by tisonkun — APPROVED** [review]
Thanks for your updates @ykgarfield! The changes look good to me.

+1 to merge.

@gosonzhang please take a look when you're free.

File: docs/tubemq_user_guide.md [hunk 1]
```diff
@@ -1,4 +1,29 @@
 # TubeMQ User Guide
+## Prerequisites
+
+- Java 7/8(Java 9+ are not test)
+- Maven
+- [protoc 2.5.0](https://github.com/protocolbuffers/protobuf/releases/tag/v2.5.0)
+
+## Install protoc
```
Comment (yiheng) [inline docs/tubemq_user_guide.md]: I suggest removing this ## Install protoc section. One concern is the instructions are only for Windows; another concern is protoc is just prerequisites as others. People should be able to find how to install protoc through Internet. That should be enough we list it as an item of prerequistes.

File: docs/tubemq_user_guide.md [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+Install on windows operating system is simple:
+
+- Unzip `protoc-2.5.0-win32.zip`, will get `protoc.exe`.
+- Config environment variable, add path of `protoc.exe` to `Path` variable.
+
+Others os can following the protobuf official docuement.
+
+## Build source code
```
Comment (yiheng) [inline docs/tubemq_user_guide.md]: Can we merge this section with the below one? `Build source code` and `Build` looks like duplicated. IMHO, It should be a supplement of the build section.

Maybe change it to
## Build
Go to the project root, and run
```bash
mvn clean package -Dmaven.test.skip
```
Each module of the project can also be build separately.

If want to build and debug source code in IDE, go to the project root, and run

```bash
mvn compile
```

This command will generate `.proto`  files corresponding Java source files(location is `target/generated-sources`).

When this command finished, you can use IDE import the project as maven project.

File: docs/tubemq_user_guide.md (hunk 1 above)
Comment (ykgarfield) [reply docs/tubemq_user_guide.md]: It's a good suggestion, I will remove

File: docs/tubemq_user_guide.md (hunk 2 above)
Comment (ykgarfield) [reply docs/tubemq_user_guide.md]: Yes, at that time, I also think these two sections are a bit duplicated.Can the `Build` section divided into two child sections.One is `Build distribution tarball`, the others is `Build source code`, this is a bit clearer.

Comment (yiheng) [reply docs/tubemq_user_guide.md]: Yes. I'm okay with that

File: docs/tubemq_user_guide.md [hunk 3]
```diff
@@ -1,11 +1,30 @@
 # TubeMQ User Guide
+## Prerequisites
+
+- Java 7/8(Java 9+ are not test)
```
Comment (yiheng) [inline docs/tubemq_user_guide.md:4]: tested?

File: docs/tubemq_user_guide.md [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+
+### Build distribution tarball
 Go to the project root, and run
 ```bash
 mvn clean package -Dmaven.test.skip
 ```
 Each module of the project can also be build separately.
 
+### Build source code
+If want to build and debug source code in IDE, go to the project root, and run
```
Comment (yiheng) [inline docs/tubemq_user_guide.md]: If you want to...?

File: docs/tubemq_user_guide.md [hunk 5]
```diff
@@ -1,11 +1,30 @@
...
+
+```bash
+mvn compile
+```
+
+This command will generate `.proto`  files corresponding Java source files(location is `target/generated-sources`).
```
Comment (yiheng) [inline docs/tubemq_user_guide.md]: corresponding to Java source files...?

**Comment by yiheng** [comment]
Thanks for your contribution. Can you also fix some small typos? Others look good to me.

**Pull request merged**